src/app/challenges/challenges.service.js

```diff
@@ -380,6 +381,33 @@ angular.module('supportAdminApp')
         return deferred.promise;
       };
 
+      /**
+       * gets a list of e-mails based on a list of users
+       * @returns {Promise} the promise with a list of userIds and e-mails.
+       */
+      function getResourceEmails(users) {
+        var deferred = $q.defer();
+
+        var qs = users.reduce(function (currentString, user, i) {
+          if (i != 0)
+            currentString += "&";
+          return currentString + "userIds[]=" +user.memberId;
+        },"");
+
+        $http({
+          method: 'GET',
+          url: MEMBER_V5_API_URL + '/members?' + qs + '&fields=userId,email&perPage=' + users.length,
```

Comment:
Instead of using `reduce` and manual string concatenation, could we use `$httpParamSerializer` https://code.angularjs.org/1.4.0/docs/api/ng/service/$httpParamSerializer to convert an object to a query string?

Reply:
using $httpParamSerializer, just a quick note that we must differentiate the parameters when getting info from a single user or multiple users